### PR TITLE
X.H.UrgencyHook: Default for DzenUrgencyHook and UrgencyConfig

### PR DESCRIPTION
**Pull request by ilya-bobyr**
As the XMonad config is commonly customized by saying

    def { startupHook = ...
        , manageHook = ...
        , ...
        }

It seems consistent to allow the same for an individual hook config:

    let urgencyHook = def { suppressWhen = ...
                          , remindWhen = ...
                          }

### Description

Include a description for your changes, including the motivation
behind them.

### Checklist

  - [x] I've read [CONTRIBUTING.md](https://github.com/xmonad/xmonad/blob/master/CONTRIBUTING.md)

  - [x] I've considered how to best test these changes (property, unit,
        manually, ...) and concluded: this is configuration only change, if it compiles, it is good.

  - [x] I updated the `CHANGES.md` file

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -191,6 +191,11 @@
     - Added an EDSL—particularly geared towards programs like terminals
       or Emacs—to spawn processes from XMonad in a compositional way.
 
+  * `XMonad.Hooks.UrgencyHook`
+
+    - Added an `Default` instance for `UrgencyConfig` and
```
Comment (slotThe) [inline CHANGES.md:196]:
```suggestion
     - Added a `Default` instance for `UrgencyConfig` and
```

File: XMonad/Hooks/UrgencyHook.hs [hunk 2]
```diff
@@ -254,7 +254,13 @@ minutes secs = secs * 60
 -- Use a variation of this in your config just as you use a variation of
 -- 'def' for your xmonad definition.
 urgencyConfig :: UrgencyConfig
-urgencyConfig = UrgencyConfig { suppressWhen = Visible, remindWhen = Dont }
+urgencyConfig = def
+
+-- | The default 'UrgencyConfig'. suppressWhen = Visible, remindWhen = Dont.
+-- Use a variation of this in your config just as you use a variation of
```
Comment (slotThe) [inline XMonad/Hooks/UrgencyHook.hs:260]: We probably want to at least put these in a monospace font (between `@`s)

**Comment by slotThe** [comment]
Additionally, we probably now want to deprecate `urgencyConfig` and `dzenUrgencyHook` (or would that be too annoying for users?  I don't know)

File: CHANGES.md (hunk 1 above)
Comment (ilya-bobyr) [reply CHANGES.md]: Done

File: XMonad/Hooks/UrgencyHook.hs (hunk 2 above)
Comment (ilya-bobyr) [reply XMonad/Hooks/UrgencyHook.hs:260]: Wrapped `suppressWhen = Visible, remindWhen = Dont` in `@`s.
Is this what you had in mind?

**Comment by ilya-bobyr** [comment]
> Additionally, we probably now want to deprecate `urgencyConfig` and `dzenUrgencyHook` (or would that be too annoying for users? I don't know)

I do not have a strong opinion, though I also do not have a lot of Haskell experience %)
It does seem reasonable to deprecate `urgencyConfig` in favor of the `Default` instance.
Not sure why was `Default` was not used in the first place.  I assumed there might be a reason I was not aware of.

As for `dzenUrgencyHook` - I do not think it should be deprecated.
There could be other urgency hook handlers.  And `borderUrgencyHook` and `spawnUrgencyHook` are already defined.
Most code will probably be more readable if it is using `dzenUrgencyHook` directly, instead of the `def` instance.

In short, I could add deprecated to `urgencyConfig`.
Having one way to do simple things is good :)

File: XMonad/Hooks/UrgencyHook.hs (hunk 2 above)
Comment (slotThe) [reply XMonad/Hooks/UrgencyHook.hs:260]: Yes exactly; Haddock will print this in a monospaced font, which I think looks nicer for code

**Comment by slotThe** [comment]
> In short, I could add deprecated to `urgencyConfig`. Having one way to do simple things is good :)

This seems fine to me; let's do it!

**Comment by ilya-bobyr** [comment]
> > In short, I could add deprecated to `urgencyConfig`. Having one way to do simple things is good :)
> 
> This seems fine to me; let's do it!

Added :)

**Comment by slotThe** [comment]
Thank you!
